test/TestPercentageMath.sol

```diff
@@ -82,18 +82,21 @@ contract TestPercentageMath is Test {
         assertEq(PercentageMath.percentAdd(x, 0), x);
     }
 
-    function testPercentAdd(uint256 x, uint256 y) public {
-        vm.assume(y <= MAX_UINT256 - PERCENTAGE_FACTOR);
-        vm.assume(x <= MAX_UINT256 / (PERCENTAGE_FACTOR + y));
+    function testPercentAdd(uint256 x, uint256 p) public {
+        vm.assume(p <= MAX_UINT256 - PERCENTAGE_FACTOR);
+        vm.assume(x <= MAX_UINT256 / (PERCENTAGE_FACTOR + p));
 
-        assertApproxEqAbs(PercentageMath.percentAdd(x, y), (x * (PERCENTAGE_FACTOR + y)) / PERCENTAGE_FACTOR, 1);
+        assertEq(
+            PercentageMath.percentAdd(x, p),
+            (x * (PERCENTAGE_FACTOR + p) + HALF_PERCENTAGE_FACTOR) / PERCENTAGE_FACTOR
+        );
```

Comment:
I would write those computations in a ref lib for maintainability

Reply:
I used mathRef